### PR TITLE
cluster: skip sig verification for versions earlier than v1.3.0

### PR DESCRIPTION
**Pull request by dB2510**
Skip EIP712 signature verification for versions earlier than v1.3.0.

category: feature
ticket: #1203

### CODE REVIEW

File: cluster/version.go [hunk 1]
```diff
@@ -53,7 +51,7 @@ func isJSONv1x3(version string) bool {
 }
 
 // SupportedVersionsForT returns the supported definition versions for testing purposes only.
-func SupportedVersionsForT(*testing.T) []string {
```
Comment (corverroos) [inline cluster/version.go:56]: please do not remove this, it enforces use in testing only

File: cluster/cluster_test.go [hunk 2]
```diff
@@ -64,6 +64,14 @@ func TestEncode(t *testing.T) {
 			)
 			require.NoError(t, err)
 			definition.Version = version
+
+			if version == "v1.0.0" || version == "v1.1.0" || version == "v1.2.0" {
```
Comment (corverroos) [inline cluster/cluster_test.go:68]: add comment why

File: cluster/definition.go [hunk 3]
```diff
@@ -140,6 +140,11 @@ func (d Definition) NodeIdx(pID peer.ID) (NodeIdx, error) {
 
 // VerifySignatures returns true if all config signatures are fully populated and valid. A verified definition is ready for use in DKG.
 func (d Definition) VerifySignatures() error {
+	// Skip signature verification for definition versions earlier than v1.3 since there are no EIP712 signatures before v1.3.0.
+	if !supportEIP712Sigs(d.Version) {
+		return nil
```
Comment (corverroos) [inline cluster/definition.go:145]: add `TODO: maybe error if signatures present`

File: cluster/definition.go [hunk 4]
```diff
@@ -437,6 +451,10 @@ func unmarshalDefinitionV1x2or3(data []byte) (def Definition, err error) {
 	return def, nil
 }
 
+func supportEIP712Sigs(version string) bool {
```
Comment (corverroos) [inline cluster/definition.go:454]: add godoc

**Review by corverroos — APPROVED** [review]

File: cluster/definition.go (hunk 3 above)
Comment (dB2510) [reply cluster/definition.go:145]: we error in UnmarshalJSON though

Comment (corverroos) [reply cluster/definition.go:146]: sure, but we should also maybe error here. If I create a old version with signatures, it shouldn't be valid. On that note, we should also error when marshalling probably.

Comment (dB2510) [reply cluster/definition.go:150]: done

**Pull request merged**